Benunc/acme-gateway
# Refactor: gateway updates

Depends on https://github.com/impress-org/givewp/pull/6658

Related: https://github.com/impress-org/givewp/pull/6695, https://github.com/impress-org/givewp/pull/6699

## Description



Updates to the example gateway:
- Change Acme to Example so there's no confusion with Acme Payments
- Remove subscription logic for now
- Update offsite route method name & logic
- Add filters for `$gatewayData` param
- Add stepped comments to make things more clear
- Simplify the example code

## Affects



Everything

## Visuals



N/A

## Testing Instructions



- Install plugin / pull changes
- Activate plugin
- Enable offsite and onsite gateways
- Donate using a donation form with both gateways for one-time and recurring
- Make sure everything works as expected.

## Pre-review Checklist
